source/agora/consensus/state/ValidatorSet.d

```diff
@@ -272,6 +272,38 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Get all the current validators in ascending order of public key
+
+        Params:
+            pub_keys = will contain the public keys
+            height = the block height for which we want the active validators
+
+        Returns:
+            Return true if there was no error in getting the public keys
+
+    ***************************************************************************/
+
+    public bool getActiveValidatorPublicKeys (out PublicKey[] pub_keys, Height height)
+        @trusted nothrow
+    {
+        try
+        {
+            assumeSafeAppend(pub_keys);
```

Comment:
You must use `ref` for this to work. Because `out` will initialize it with `null`, which means `assumeSafeAppend` has no effect.

Reply:
So the following function is also wrong as it is the same code pattern? I will wait for your input before updating. Maybe other places need to be fixed also.

Reply:
Just realised that was me too :-) so will update both new functions to use ref.